cni/pkg/ambient/net.go

```diff
@@ -96,13 +111,16 @@ func addPodToMeshWithIptables(pod *corev1.Pod, ip string) {
 	dev, err := getDeviceWithDestinationOf(ip)
 	if err != nil {
 		log.Warnf("Failed to get device for destination %s", ip)
-		return
+		return err
 	}
 
 	err = disableRPFiltersForLink(dev)
 	if err != nil {
 		log.Warnf("failed to disable procfs rp_filter for device %s: %v", dev, err)
+		// I believe this is not a fatal error ?
```

Comment:
It depends on OS defaults - if this fails things may still work if the defaults do what we need, but we can't guarantee that. Arguably should be fatal for clarity.

Reply:
AFAIK rp_filter is used for IP spoofing. We should be able to operate without spoofing - Istio with sidecars is not spoofing by default, and ztunnel can still return metadata even if the address is not spoofed. 

But we might want to have user explicitly configure and treat it as a fatal error - at least in any 'on-by-default' cases, since lack of spoofing can result in app breakages.